src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs

```diff
@@ -642,32 +642,36 @@ public void ResetTo(KeccakHash original)
 
         private static class Pool
         {
-            private const int MaxPooled = 24;
-            private static readonly ConcurrentQueue<byte[]> s_remainderCache = new();
-            public static byte[] RentRemainder() => s_remainderCache.TryDequeue(out byte[]? remainder) ? remainder : new byte[STATE_SIZE];
+            private const int MaxPooledPerThread = 4;
+            [ThreadStatic]
+            private static Queue<byte[]>? s_remainderCache;
```

Comment:
Still needs queue as are area where Keccaks are duplicated with unfinalized buffers (where the start is same, then remainders are appended)